### PR TITLE
feat: Add doc for implementing new algorithms

### PR DESCRIPTION
**Pull request by gaocegege**
Signed-off-by: Ce Gao <gaoce@caicloud.io>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/769)

### CODE REVIEW

File: docs/developer-guide.md [hunk 1]
```diff
@@ -21,39 +21,256 @@
...
+            trials=Assignment.generate(list_of_assignments)
+        )
 ```
 
-You can build all images from source for v1alpha2 as follows:
+### Make the algorithm a GRPC server
 
-```bash
-make images
+Create a package under [cmd/suggestion](../cmd/suggestion). Then create the main function and Dockerfile. Here is an example: [cmd/suggestion/hyperopt](../cmd/suggestion/hyperopt). Then build the Docker image.
```
Comment (hougangliu) [inline docs/developer-guide.md]: Please also highlight the service should serve with 6789 port

Comment (gaocegege) [reply docs/developer-guide.md]: SGTM

Comment (johnugeorge) [reply docs/developer-guide.md]: @gaocegege Missed this?

**Comment by johnugeorge** [comment]
Also, Add one sentence on ValidateAlgorithmSettings API?

**Comment by johnugeorge** [comment]
And also a link to  proto file might be helpful as a reference

File: docs/developer-guide.md (hunk 1 above)
Comment (gaocegege) [reply docs/developer-guide.md]: Oh yeah. I am adding it.

**Pull request merged**